src/createRollup.ts

```diff
@@ -0,0 +1,202 @@
+  coreContracts: CoreContracts;
+};
+
+/**
+ * Performs the tx to deploy the chain's core contracts.
+ *
+ * Before creating a custom gas token chain, it checks the custom gas
+ * token allowance granted to the rollup creator contract. Runs an approval
+ * tx for insufficient allowance.
+ *
+ * Accepts rollup creation config, rollup owner, and the parent chain public client.
+ *
+ * Returns the transaction, the transaction receipt, and the core contracts.
+ *
+ * - Example 1: [Create an ETH gas token chain](https://github.com/OffchainLabs/arbitrum-orbit-sdk/blob/main/examples/create-rollup-eth/index.ts)
+ * - Example 2: [Create a custom gas token chain](https://github.com/OffchainLabs/arbitrum-orbit-sdk/blob/main/examples/create-rollup-custom-fee-token/index.ts)
+ *
+ * @param {CreateRollupFunctionParams} createRollupFunctionParams {@link CreateRollupFunctionParams}
+ * @param {Object} createRollupFunctionParams.config - The chain config
+ * @param {string} createRollupFunctionParams.batchPoster - The batchPoster address
+ * @param {Array<string>} createRollupFunctionParams.validators - The validator(s) address array
+ * @param {boolean} [createRollupFunctionParams.deployFactoriesToL2=] - Optional, defaults to true.
+ * Deploying factories via retryable tickets at rollup creation time is the most reliable method to
+ * do it since it doesn't require paying the L1 gas. If deployment is not done as part of rollup
+ * creation TX, there is a risk that anyone can try to deploy factories and potentially burn
+ * the nonce 0 (ie. due to gas price spike when doing direct child chain TX). That would mean we
+ * permanently lost capability to deploy deterministic factory at expected address.
+ * @param {string} [createRollupFunctionParams.nativeToken=] - The native token address, optional, defaults to ETH
+ * @param {number} [createRollupFunctionParams.maxDataSize=] - The max calldata size, optional, defaults to 104_857 B for Orbit chains
+ * @param {number} [createRollupFunctionParams.maxFeePerGasForRetryables=] - The max fee per gas for retryables, optional, defaults to 0.1 gwei
+ * @param {Object} createRollupFunctionParams.rollupOwner - The rollup owner private key account
+ * @param {Object} createRollupFunctionParams.parentChainPublicClient - The parent chain Viem Public Client
+ *
+ * @returns Promise<{@link CreateRollupResults}> - the transaction, the transaction receipt, and the core contracts.
+ *
+ * @example
+ * const createRollupConfig = createRollupPrepareConfig({
+ *   chainId: BigInt(chainId),
+ *   owner: deployer.address,
+ *   chainConfig: prepareChainConfig({
+ *     chainId,
+ *     arbitrum: {
+ *       InitialChainOwner: deployer.address,
+ *       DataAvailabilityCommittee: true,
+ *     },
+ *   }),
+ * });
+ *
+ * const {
+ *   transaction,
+ *   transactionReceipt,
+ *   coreContracts,
+ * } = await createRollup({
+ *   config: createRollupConfig,
+ *   rollupOwner: deployer,
+ *   batchPoster,
+ *   validators,
```

Comment:
I'm thinking we group all of the params that are actual args to the contract into one `params` field or something like that

Reply:
do you mean something like this?
```typescript
createRollup(createRollupParams, { account, parentChainPublicClient })
```

Reply:
Same as here https://github.com/OffchainLabs/arbitrum-orbit-sdk/blob/main/src/createRollupPrepareTransactionRequest.ts#L32